tests/managers/file/FileManagerTests.js

```diff
@@ -0,0 +1,472 @@
+const chai = require('chai');
+const expect = chai.expect;
+chai.use(require('chai-as-promised'));
+const SDL = require('./../../../lib/node/dist/SDL.min.js');
+
+// Mocking framework
+// Used to stub an RPC call so that it isn't actually sent to Core
+const sinon = require('sinon');
+
+const Validator = require('../../Validator');
+const Test = require('../../Test');
+
+module.exports = function (appClient) {
+    const sdlManager = appClient._sdlManager;
+    const fileManager = sdlManager.getFileManager();
+    const lifecycleManager = sdlManager._lifecycleManager;
+    const validFile = new SDL.manager.file.filetypes.SdlFile();
+    validFile.setName(Test.GENERAL_STRING);
+    validFile.setFileData(new Array(50));
+    validFile.setPersistent(false);
+
+    describe('FileManagerTests', function () {
+        /**
+         * Handle ListFiles successes.
+         * @returns {Promise} - A promise.
+         */
+        function onListFilesSuccess () {
+            const responseSuccess = new SDL.rpc.messages.ListFilesResponse({
+                functionName: SDL.rpc.enums.FunctionID.ListFiles,
+            })
+                .setFilenames(Test.GENERAL_STRING_LIST)
+                .setSpaceAvailable(Test.GENERAL_INT)
+                .setSuccess(true);
+            // _handleRpc triggers the listener
+            sdlManager._lifecycleManager._handleRpc(responseSuccess);
+
+            return new Promise((resolve, reject) => {
+                resolve(responseSuccess);
+            });
```

Comment:
No need to change it, but for future reference, there is a shorthand for immediately sending a resolved promise: `return Promise.resolve(responseSuccess);`